Source/Fuse.CameraRoll/iOS/iOSCameraRoll.uno

```diff
@@ -41,11 +41,28 @@ namespace Fuse.CameraRoll
 		[Foreign(Language.ObjC)]
 		static void SelectPictureInternal(Action<string> onComplete, Action<string> onFail)
 		@{
-			dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-				[[CameraRollHelper instance] selectPictureWithCompletionHandler:onComplete onFail:onFail];
-			});
+			PHAuthorizationStatus status = [PHPhotoLibrary authorizationStatus];
+			if (status == PHAuthorizationStatusNotDetermined)
+			{
+				[PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status)
+				{
+					if (status == PHAuthorizationStatusAuthorized)
+					{
+						dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+							[[CameraRollHelper instance] selectPictureWithCompletionHandler:onComplete onFail:onFail];
+						});
+					} else {
+						onFail(@"User cancelled");
+					}
+				}];
+			} else if (status == PHAuthorizationStatusAuthorized)
+			{
+				dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+					[[CameraRollHelper instance] selectPictureWithCompletionHandler:onComplete onFail:onFail];
+				});
+			} else {
+				onFail(@"User cancelled");
```

Comment:
I don't think "User cancelled" is accurate here. Here's the full list of return-values, taken from [here](https://developer.apple.com/documentation/photokit/phauthorizationstatus?language=objc):
- `PHAuthorizationStatusNotDetermined`: Explicit user permission is required for photo library access, but the user has not yet granted or denied such permission.
- `PHAuthorizationStatusRestricted`: Your app is not authorized to access the photo library, and the user cannot grant such permission.
- `PHAuthorizationStatusDenied`: The user has explicitly denied your app access to the photo library.
- `PHAuthorizationStatusAuthorized`: The user has explicitly granted your app access to the photo library.

So, the first and last ones are handled, but the difference between `PHAuthorizationStatusRestricted` and `PHAuthorizationStatusDenied` seems relevant here.

Perhaps we should check for either, and pass the string from the documentation instead?

Reply:
Done. It now passes a string. Maybe we should put these values in the docs so Fusers can now what they mean.